src/parser/esprima_cpp/esprima.cpp

```diff
@@ -1590,14 +1598,16 @@ class Parser {
             {
                 Value v;
                 if (builder.isNodeGenerator()) {
-                    if (token->type == Token::NumericLiteralToken) {
-                        double d = token->valueNumberLiteral(this->scanner);
-                        if (this->context->inLoop || d == 0) {
-                            this->insertNumeralLiteral(Value(d));
+                    if (builder.willGenerateByteCode()) {
```

Comment:
IMO `builder.isNodeGenerator()` can be replaced with `builder.willGenerateByteCode()` at here.
BTW if `isNodeGenerator` is no longer used how about remove this function?

Reply:
isNodeGenerator is needed
isNodeGenerator is not same with willGenerateByteCode
in small config, every builder is NodeGenerator.

Reply:
Yes, but in this case (IMHO) the upper `isNodeGenerator` still can be replaced by `willGenerateByteCode`.
In small config mode, every builder would be `NodeGenerator`, so isNodeGenerator check is not necessary.
In normal mode, since only the target builder (also NodeGenerator) would have true value for `willGenerateByteCode`, just using `willGenerateByteCode` is enough.
So using `isNodeGenerator` and `willGenerateByteCode` together seems redundant here.
(correct me if I'm wrong)

Reply:
if I don't check isNodeGenerator, `builder.setValueStringLiteral(token->valueStringLiteral(this->scanner));` is evaluated
but NodeGenerator don't have setValueStringLiteral method

Reply:
Oh, I fully understand it. Thanks!